File: sliding_window/209_min_size_subarray.py
def minSubArrayLen(target: int, nums: list[int]) -> int:
    begin = 0
    window_state = 0
    result = float("inf")

    for end in range(len(nums)):
        window_state += nums[end]

        while window_state >= target:
            window_size = end - begin + 1
            result = min(result, window_size)
            window_state -= nums[begin]
            begin += 1

    if result == float("inf"):
        return 0

    return result

print(minSubArrayLen(3, [1, 1, 2, 1, 3, 1, 1]))

File: sliding_window/209_min_size_subarray.go
import "math"

func minSubArrayLen(target int, nums []int) int {
	begin := 0
	result := math.Inf(1) // float64
	windowState := 0

	for end := range len(nums) {
		windowState += nums[end]

		for windowState >= target {
			windowSize := end - begin + 1
			result = min(result, float64(windowSize)) // приведение типа
			windowState -= nums[begin]
			begin++
		}
	}

	if result == math.Inf(1) {
		return 0
	}

	return int(result) // приведение типа
}